.github/workflows/yocto-build-deploy.yml

```
@@ -593,28 +600,28 @@ jobs:
 
       - name: Set S3 ACL (private)
         id: s3-acl-private
-        if: inputs.deploy-s3 == true && steps.balena-lib.outputs.is_private != 'false'
+        if: steps.should-deploy.outputs.deploy == true && steps.balena-lib.outputs.is_private != 'false'
```

```suggestion
        if: steps.should-deploy.outcome == 'success' && steps.balena-lib.outputs.is_private != 'false'
```
Since these outputs are always true when the step conditionally runs, another way to check the value is via the [outcome of the step](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/contexts#steps-context) (and the step can simply exit 0 rather than set an output).

Up to you which you feel is more clean